classes/publication/Repository.php

```diff
@@ -228,19 +228,20 @@ public function edit(Publication $publication, array $params)
 
         parent::edit($publication, $params);
 
-        // Create or delete the thumbnail of a cover image
-        if (array_key_exists('coverImage', $params)) {
-            $publicFileManager = new PublicFileManager();
-            $submission = Repo::submission()->get($publication->getData('submissionId'));
+        $updatedPublication = $this->get($publication->getId());
+        $coverImages = $updatedPublication->getData('coverImage');
 
+        // Create or delete the thumbnail of a cover image
+        $publicFileManager = new PublicFileManager();
+        $submission = Repo::submission()->get($publication->getData('submissionId'));
+        if ($coverImages) {
             // Get the submission context
             $submissionContext = $this->request->getContext();
             if ($submissionContext->getId() !== $submission->getData('contextId')) {
                 $submissionContext = Services::get('context')->get($submission->getData('contextId'));
             }
 
-            foreach ($params['coverImage'] as $localeKey => $newCoverImage) {
-
+            foreach ($coverImages as $localeKey => $newCoverImage) {
                 // Delete the thumbnail if the cover image has been deleted
                 if (is_null($newCoverImage)) {
```

Comment:
I think this will never be executed, because there is no $localeKey in $coverImages that points to null or an empty array. The $coverImages seems to always contain only existing images, for existing $localeKey.
Thus, if we remove a cover image from only one locale and leave the cover image for the other locale, that thumbnail will not be removed.